### PR TITLE
Fix linter issues

### PR DESCRIPTION
**Pull request by thardeck**

### CODE REVIEW

File: pkg/bundle/resources.go [hunk 1]
```diff
@@ -120,7 +120,8 @@ func chartURL(location *fleet.HelmOptions, auth Auth) (string, error) {
 		pool.AppendCertsFromPEM(auth.CABundle)
 		transport := http.DefaultTransport.(*http.Transport).Clone()
 		transport.TLSClientConfig = &tls.Config{
-			RootCAs: pool,
+			RootCAs:    pool,
+			MinVersion: tls.VersionTLS12,
```
Comment (manno) [inline pkg/bundle/resources.go:124]: I hope all clients are ok with that. Who uses that endpoint, k8s kubelet or fleet itself?

File: pkg/controllers/bundle/controller.go [hunk 2]
```diff
@@ -101,7 +101,10 @@ func (h *handler) OnClusterChange(_ string, cluster *fleet.Cluster) (*fleet.Clus
 		}
 		for _, bundleDeployment := range bundleDeployments {
 			logrus.Debugf("cleaning up bundleDeployment %v in namespace %v not matching the cluster: %v", bundleDeployment.Name, bundleDeployment.Namespace, cluster.Name)
-			h.bundleDeployments.Delete(bundleDeployment.Namespace, bundleDeployment.Name, nil)
+			err := h.bundleDeployments.Delete(bundleDeployment.Namespace, bundleDeployment.Name, nil)
```
Comment (manno) [inline pkg/controllers/bundle/controller.go:104]: Hm, `_ := h.bundleDeployments.Delete` is closer to the original behavior. Maybe this error is not to be checked and it is ok for bundles to not delete... 🤷 

Nah, better to check this and see if tests pass.

File: cmd/fleetagent/main.go [hunk 3]
```diff
@@ -4,7 +4,6 @@ import (
 	"fmt"
 	"log"
 	"net/http"
-	_ "net/http/pprof"
```
Comment (manno) [inline cmd/fleetagent/main.go:7]: If we remove that, we can remove the whole http server in line 29

File: cmd/fleetcontroller/main.go [hunk 4]
```diff
@@ -3,7 +3,6 @@ package main
 import (
 	"log"
 	"net/http"
-	_ "net/http/pprof"
```
Comment (manno) [inline cmd/fleetcontroller/main.go:6]: same as above, might as well remove the whole server, or guard it with a shell variable.

File: pkg/bundle/resources.go (hunk 1 above)
Comment (thardeck) [reply pkg/bundle/resources.go:124]: As discussed, TLS was introduced 2008, is also the Minimum in modern browsers like Firefox without know issues, so it should be fine.
We will document it in the next release notes though.

File: pkg/controllers/bundle/controller.go (hunk 2 above)
Comment (thardeck) [reply pkg/controllers/bundle/controller.go:104]: `h.bundleDeployments.Delete` is a wrapper around client go delete so I suppose we properly want to handle errors. If this change results in issues we should fix them.

File: pkg/bundle/resources.go [hunk 5]
```diff
@@ -360,6 +362,10 @@ func readContent(ctx context.Context, progress *progress.Progress, base, name st
 	}
 
 	err = filepath.Walk(temp, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
```
Comment (manno) [inline pkg/bundle/resources.go:365]:
```
// The err argument reports an error related to path, signaling that Walk
// will not walk into that directory. The function can decide how to
// handle that error; as described earlier, returning the error will
// cause Walk to stop walking the entire tree.
```
Returning that error will cause walk to abort if one of the paths is not accessible. That changes the existing behavior of the function. If I understand correctly this func is used to read bundle resources, so we probably do want to abort if part of the bundle is unreadable. However we need to document this change.

**Pull request merged**